Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -299,11 +299,21 @@ function Wait-ServiceFabricApplicationUpgradeAction
     }
 
     $upgradeStatusValidator = { param($upgradeStatus) return !($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted") }
-    $upgradeStatus = Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
-        -ResultRetryEvaluator $upgradeStatusValidator `
-        -MaxTries 2147483647 `
-        -RetryIntervalInSeconds 5 `
-        -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+
+    try
+    {
+        $upgradeStatus = Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
+            -ResultRetryEvaluator $upgradeStatusValidator `
+            -MaxTries 2147483647 `
+            -RetryIntervalInSeconds 5 `
+            -RetryableExceptions @("System.Fabric.FabricTransientException", "System.TimeoutException")
+    }
+    catch
+    {
```

Comment:
This will print health when failure to get upgrade status. This is not needed. We should print health when upgrade itself failed. i.e. if upgrade was rolled back.

Reply:
Isn' t it okay to print in more cases than required?

Reply:
it might be okay. this error will happen when task fails to get application upgrade status - may be due to transient exception or due to timeout. but health should definitely be printed when upgrade is rolled back.